packages/dynamite/dynamite/lib/src/builder/resolve_type.dart

```diff
@@ -145,7 +145,7 @@ TypeResult _resolveType(
 
     case openapi.Schema(type: openapi.SchemaType.array):
       final TypeResult subResult;
-      if (schema.maxLength == 0) {
+      if (schema.maxItems == 0) {
```

Comment:
Please remove this whole commit for now. Patching all the specs is not feasible.
We will just have to wait until https://github.com/nextcloud/neon/pull/1894 and accept that both dynamite and the current specs have this problem.

Reply:
Then this PR has to wait.

I will not mixin the `StringValidator` into the `ArraySchema` just to reproduce this issue with the latest commit.

Reply:
You're right. Another option would be to remove the first and the last commit and keep everything else as that can be merged easily.

Reply:
I'm not sure it is worth it.
We do not rely on these cleanups so they can just wait (unless you would like to have some of them pulled in already).

Reply:
I mean cleanups are good, but if it blocks no other work and just creates additional work now, we can just wait.

Reply:
I just rebased onto #1894.
Should only be a couple of days